modules/apps/batch-engine/batch-engine-service/src/main/java/com/liferay/batch/engine/internal/reader/BatchEngineImportTaskItemReaderUtil.java

```diff
@@ -136,6 +138,36 @@ public static Map<String, Object> mapFieldNames(
 						targetFieldName, entry.getValue());
 				}
 			}
+			else {
+				String[] fieldNameParts = StringUtil.split(
+					entry.getKey(), StringPool.PERIOD);
+
+				targetFieldName = (String)fieldNameMappingMap.get(
+					fieldNameParts[0]);
+
+				if (Validator.isNotNull(targetFieldName)) {
+					if (fieldNameParts[1].matches(
+							_MULTISELECT_PICKLIST_KEY_REGEX)) {
```

Comment:
I was also thinking along the same lines, but the format of serialized (Multiselect)Picklist was decided by the Product side here https://liferay.slack.com/archives/C03U66A6NNN/p1706189788474469. Because of the timeline I suggest we leave this discussion for another time.